src/state.rs

```diff
@@ -456,39 +614,61 @@ impl DistributedCtx<LunaticEnvironment> for DefaultProcessState {
         runtime: WasmtimeRuntime,
         module: Arc<WasmtimeCompiledModule<Self>>,
         config: Arc<Self::Config>,
+        // tracer: Arc<BoxedTracer>,
+        // tracer_context: Arc<Context>,
+        // logger: Arc<Logger>,
     ) -> Result<Self> {
         let signal_mailbox = unbounded_channel();
         let signal_mailbox = (signal_mailbox.0, Arc::new(Mutex::new(signal_mailbox.1)));
         let message_mailbox = MessageMailbox::default();
+        let id = environment.get_next_process_id();
+        let wasi = build_wasi(
+            Some(config.command_line_arguments()),
+            Some(config.environment_variables()),
+            config.preopened_dirs(),
+        )?;
         let state = Self {
-            id: environment.get_next_process_id(),
+            id,
             environment,
             distributed: Some(distributed),
             runtime: Some(runtime),
             module: Some(module),
-            config: config.clone(),
+            config,
             message: None,
             signal_mailbox,
             message_mailbox,
             resources: Resources::default(),
-            wasi: build_wasi(
-                Some(config.command_line_arguments()),
-                Some(config.environment_variables()),
-                config.preopened_dirs(),
-            )?,
+            wasi,
             wasi_stdout: None,
             wasi_stderr: None,
             initialized: false,
             registry: Default::default(), // TODO move registry into env?
             db_resources: DbResources::default(),
             registry_atomic_put: None,
+            tracer: Arc::new(BoxedTracer::new(Box::new(NoopTracer::new()))),
+            tracer_context: Arc::new(Context::new()),
+            process_context: Context::new(),
+            meter_provider: GlobalMeterProvider::new(NoopMeterProvider::new()),
+            logger: Arc::new(
+                env_logger::Builder::new()
+                    .filter_level(log::LevelFilter::Off)
+                    .build(),
+            ),
```

Comment:
I haven't fully understood about new_dist_state, and have made all logging for this be noops.

Is there any advice on what I should do here? Should logging to the terminal print on the control server?

Reply:
So dist state only has references to control client and "node". Node implements communication with other nodes (for example, it receives a spawn command from another node). Control client talks to the control server.

We can leave it out for now.